test/Npgsql.Tests/ExceptionTests.cs

```diff
@@ -308,14 +293,26 @@ public void Serialization()
             Assert.That(expected.Routine, Is.EqualTo(actual.Routine));
         }
 
-        SerializationInfo CreateSerializationInfo()
+        SerializationInfo CreateSerializationInfo() => new SerializationInfo(typeof(PostgresException), new FormatterConverter());
+#pragma warning restore 618
+#pragma warning restore SYSLIB0011
+
+        [Test]
+        [IssueLink("https://github.com/npgsql/npgsql/issues/3204")]
+        public void BaseExceptionPropertySerialization()
```

Comment:
Any specific reason this needs to be a separate test, rather than just testing these properties in the big one above?

Reply:
Just that it needed to test the values in the SerializationInfo object directly, which is different to what the Serialization test above does - compare properties of the original and deserialized exception. (That passes without the fix.) So it's kind of testing two different things in different ways, though they're both serialization-related.